src/core/bpf-firewall.c

```diff
@@ -651,31 +651,24 @@ int bpf_firewall_load_custom(Unit *u) {
 }
 
 static int attach_custom_bpf_progs(Unit *u, const char *path, int attach_type, Set **set, Set **set_installed) {
-        BPFProgram *prog;
+        _cleanup_(bpf_program_freep) BPFProgram *prog;
         int r;
 
         assert(u);
 
-        set_clear(*set_installed);
+        set_clear_free(*set_installed);
+        *set_installed = set_new(NULL);
```

Comment:
This is leaking. You need to define a set with the hash op so it uses the proper free function. You want `set_new(&bpf_program_hash_ops)` instead